R/serialized_tests.R

```diff
@@ -0,0 +1,116 @@
+#' Run a collection of tests
+#'
+#' Run transformations on all *-in.R files in a test directory and compare them
+#'   with their *-out.R counterpart.
+#' @param test The test to run. It corresponds to a folder name in
+#'   tests/testthat.
+#' @param sub_test A regex pattern to further reduce the amount of test files
+#'   to be tested in the test. `sub_test` must match the beginning of file
+#'   names in tests/testthat. `NULL` matches all files.
+#' @details Each file name that matches `test` and `sub_test` and ends with
+#'   "-in.R" is considered as an input to test. Its counterpart,
+#'   the reference to compare it against is the *-out.R file. It is constructed
+#'   by taking the substring of the *-in.R file before the
+#'   first dash and adding -out.R. This allows for multiple in.R files to
+#'   share one out.R file. You could have one_line-out.R as the reference to
+#'   compare one_line-random-something-stuff-in.R and
+#'   one_line-random-but-not-so-much-in.R.
+#'
+#'   This also implies that -out.R files cannot have more than one dash in
+#'   their name, i.e. just the one before out.R.
+#' @inheritParams transform_and_check
+#' @importFrom purrr flatten_chr pwalk
+test_collection <- function(test, sub_test = NULL,
+                            write_back = FALSE,
+                            transformer) {
+
+  path <- if_else(interactive(),
+                  file.path("tests", "testthat", test),
+                  file.path(test))
+
```

Comment:
@krlmlr This path creation is suboptimal, I know, but unfortunately, `here` did not work here, maybe because during R CMD check, the test folder is placed somewhere else? `devtools::check()` succeeded, but `R CMD check` not. Maybe you know a better way. Rather than `interactive()`, it should be soemthing like `r_cmd_check_is_running()` but I could not find such a function. Maybe we can make it dependent on some global variables `R CMD check` sets, but I have tool little knowledge about that.